docs/dev/alternative_release_image_sources.md

```diff
@@ -0,0 +1,161 @@
+}
+```
+
+If the release-image `q.io/ocp/release-x.y@sha256:abc` which has references to the images in repositories `q.io/openshift/x.y` was mirrored to `local.registry.com/ocp/release-x.y`, the `install-config.yaml` would look like,
+
+```yaml
+...
+imageContentSources:
+- sources:
+  - local.registry.com/ocp/release-x.y
+  - q.io/ocp/release-x.y
+  - q.io/openshift/x.y
+...
+```
+
+### ImageContentSourcePolicy
+
+If a list of `ImageContentSources` is specified, the installer configures the [RepositoryDigestMirrors][repository-digest-mirrors] for each `ImageContentSource`.
+
+If the `install-config.yaml` provided by the user is:
+
+```yaml
+...
+imageContentSources:
+- sources:
+  - local.registry.com/ocp/release-x.y
+  - q.io/ocp/release-x.y
+  - q.io/openshift/x.y
+...
+```
+
+The `ImageContentSourcePolicy` object would look like:
+
+```yaml
+...
+repositoryDigestMirrors:
+- sources:
+  - local.registry.com/ocp/release-x.y
+  - q.io/ocp/release-x.y
+  - q.io/openshift/x.y
+...
+```
+
+### release-image location
+
+The release-image location that is propagated to the bootstrap node and the cluster-version-operator will continue to be the embedded release-image location.
+
+### Bootstrap machine containers-registries.conf
+
+If a list of `ImageContentSources` is specified, the [Registries][registry-containers-registries-conf] will be configured to have each source be a mirror for another.
+
+For example,
+
+If the `install-config.yaml` provided by the user is:
+
+```yaml
+...
+imageContentSources:
+- sources:
+  - local.registry.com/ocp/release-x.y
+  - q.io/ocp/release-x.y
+  - q.io/openshift/x.y
+...
+```
+
+```toml
+[[registry]]
+location = "local.registry.com/ocp/release"
+mirror-by-digest-only = true
+
+[[registry.mirror]]
+location = "local.registry.com/ocp/release"
+
+[[registry.mirror]]
+location = "q.io/ocp/release-x.y"
+
+[[registry.mirror]]
+location = "q.io/openshift/x.y"
+
+[[registry]]
+location = "q.io/ocp/release-x.y"
+mirror-by-digest-only = true
+
+[[registry.mirror]]
+location = "local.registry.com/ocp/release"
+
+[[registry.mirror]]
+location = "q.io/ocp/release-x.y"
+
+[[registry.mirror]]
+location = "q.io/openshift/x.y"
+
+[[registry]]
+location = "q.io/openshift/x.y"
+mirror-by-digest-only = true
+
+[[registry.mirror]]
+location = "local.registry.com/ocp/release"
+
+[[registry.mirror]]
+location = "q.io/ocp/release-x.y"
+
+[[registry.mirror]]
+location = "q.io/openshift/x.y"
+```
+
+### oc adm release mirror
+
+The `release mirror` mirrors the a release-image and all other images referenced in the release-image to another repository and then provides user's details of setting up the `install-config.yaml` for the user.
+
+The output would look like:
+
+```console
+Release Image q.io/ocp/release-x.y@sha256:abcd was successfully mirrored to local.registry.com/ocp/release-x.y@sha256:abcd
+
+Following section can be added to the install-config.yaml to create a cluster using new repository:
+imageContentSources:
```

Comment:
Should this be `imageContentSource:`?  I'm seeing both in this code and I need to align with my PR for `oc`. https://github.com/openshift/origin/pull/23381

Reply:
i think it needs to be `imageContentSources` based on the json tag on the actual install-config
https://github.com/openshift/installer/blame/bea51930a0e49b5da2fb58f2184ba27e79ceeb6c/docs/dev/alternative_release_image_sources.md#L29

Reply:
Thanks!  I'll update the mirror command.